core/controllers/topic_editor.py

```diff
@@ -149,8 +151,45 @@ def get(self, topic_id: str) -> None:
         additional_story_summary_dicts = [
             summary.to_dict() for summary in additional_story_summaries]
 
+        canonical_stories_ids = [summary['id'] for
+            summary in canonical_story_summary_dicts]
+        canonical_stories = story_fetchers.get_stories_by_ids(
+            canonical_stories_ids)
+        non_empty_canonical_stories = [story for story in
+            canonical_stories if story is not None]
         updated_canonical_story_summary_dicts = []
         for summary in canonical_story_summary_dicts:
+            story = None
+            for canonical_story in non_empty_canonical_stories:
+                if canonical_story.id == summary['id']:
+                    story = canonical_story
+            if story is None:
+                continue
+            nodes = story.story_contents.nodes
+            total_chapters_count = len(nodes)
+            published_chapters_count = 0
+            upcoming_chapters_count = 0
+            overdue_chapters_count = 0
+            upcoming_chapters_expected_days = []
+            for node in nodes:
+                if node.status == constants.STORYNODE_STATUS_PUBLISHED:
+                    published_chapters_count += 1
+                if (node.status != constants.STORYNODE_STATUS_PUBLISHED and
+                    node.planned_publication_date is not None):
+                    current_time = (
+                        utils.convert_millisecs_time_to_datetime_object(
+                        utils.get_current_time_in_millisecs() -
```

Comment:
No it is not needed here, it is managed in the frontend.

Reply:
I mean this is timezone aware because you are adjusting it using `time.timezone`. This code runs in the server. I want to ensure that the timezone used here and in `node.planned_publication_date` is the same.

Reply:
Ok, I checked it. node.planned_publication_date is timezone dependent, so the time.timezone() offset has to be written. That is, in the current code, both have same timezones.

Reply:
Can you point me to the place where the value in `node.planned_publication_date` is first set?

Reply:
It is set [here](https://github.com/oppia/oppia/blob/517bf2b57072f46acc36d2aa007b844f808735e2/core/domain/story_domain.py#L729) and the time value of planned_publication_date_msecs is taken in the frontend using Date.now()

Reply:
This is wrong then. `Date.now` always returns utc timestamp but you are comparing it with a timezone adjusted timestamp

Reply:
Actually it is not wrong. Date.now returns timestamp (which is never timezone dependent). In the link above, where node.planned_publication_date is being set from planned_publication_date_msecs, the implementation used is datetime.fromtimestamp(), and you can see [here](https://docs.python.org/3/library/datetime.html#datetime.date.fromtimestamp) that it gives local timezone datetime. So while comparing here too, I have taken the time.timezone factor in the current time.

I have also verified this by actually running the server, and checked the values of node.planned_publication_date and current_time, and both are having the timezone offset.